lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/natayaprice/CsvDataLoader.java

```diff
@@ -0,0 +1,115 @@
+package com.codedifferently.lesson12.factory.natayaprice;
+
+import com.codedifferently.lesson12.factory.LibraryCsvDataLoader;
+import com.codedifferently.lesson12.models.CheckoutModel;
+import com.codedifferently.lesson12.models.LibraryDataModel;
+import com.codedifferently.lesson12.models.LibraryGuestModel;
+import com.codedifferently.lesson12.models.MediaItemModel;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CsvDataLoader implements LibraryCsvDataLoader {
+  @Override
+  public LibraryDataModel loadData() {
+    var model = new LibraryDataModel();
+    model.mediaItems = new ArrayList<>(); // You load this from the file
+    model.guests = new ArrayList<>(); // You also load this from the file.
+
+    // STEP 1: Load each file into it's own collection
+    List<MediaItemModel> mediaItems =
+        readMediaItems(
+            "/workspaces/NatayaPcode-differently-24-q1/lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/models/MediaItemModel.java");
```

Comment:
You need to pull these files from a project relative place (the test runner on GItHub doesn't have access to your laptop). Consider using `ClassPathResource` to do this or take a look at my `LibraryJsonDataLoader` as an example.